low-level/files/opt/seagate/sspl/setup/resource_map/server.py

```diff
@@ -234,4 +236,27 @@ def prepare_mem_json(self, status, description):
         self.set_health_data(
             memory_dict, status=status, description=description,
             specifics=specifics)
-        return memory_dict
\ No newline at end of file
+        return memory_dict
+
+    def get_raid_array_info(self):
+        raid_array = RaidArray()
+        raid_array_data = []
+        for raid in raid_array.get_configured_devices():
+            raid_data = self.get_health_template(raid.split("/")[-1], False)
+            raid_data["last_updated"] = int(time.time())
+            health = raid_array.get_raid_health(raid)
+            data_integrity_status = "NA"
+            devices = []
+            if health != "Missing":
+                data_integrity_status = raid_array.get_data_integrity_status(raid)
+                devices = raid_array.get_devices_status(raid)
+            specifics = [{"location": raid, "data_integrity_status" : data_integrity_status, "devices": devices}]
```

Comment:
data_integrity_status  will have true or false here, doesnt give clear idea. need better name
raid_integrity_error
Also we should give mismatch count as part of specifics
raid_integrity_mismatch_count

Reply:
Done